src/main/groovy/com/athaydes/gradle/ceylon/task/FatJarTask.groovy

```diff
@@ -0,0 +1,46 @@
+package com.athaydes.gradle.ceylon.task
+
+import com.athaydes.gradle.ceylon.CeylonConfig
+import com.athaydes.gradle.ceylon.util.CeylonCommandOptions
+import com.athaydes.gradle.ceylon.util.CeylonRunner
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+@CompileStatic
+class FatJarTask extends DefaultTask {
+
+    static List inputFiles( Project project, CeylonConfig config ) {
+        [ project.buildFile,
+          project.files( config.sourceRoots, config.resourceRoots ),
+          project.tasks.withType( GenerateOverridesFileTask ) ]
+    }
+
+    static File outputDir( Project project, CeylonConfig config ) {
+        project.file( config.output )
```

Comment:
The only output of this task is a fat jar. So this method should only point at the jar (or at least the directory the jar will be copied to).